### PR TITLE
[WIP] feat: add dailyChallenges 

### PR DESCRIPTION
**Pull request by dhruv1294**

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -109,6 +109,9 @@ dependencies {
 
     //Dagger
     implementation (Libs.dagger)
+    implementation (Libs.kotlinStd)
+    implementation "org.jetbrains.kotlin:kotlin-stdlib:$kotlin_version"
+    implementation 'androidx.legacy:legacy-support-v4:1.0.0'
```
Comment (ranaaditya) [inline app/build.gradle:114]: use buildSrc to define dependencies !

File: build.gradle [hunk 2]
```diff
@@ -1,6 +1,10 @@
 // Top-level build file where you can add configuration options common to all sub-projects/modules.
 buildscript {
 
+
+    ext {
```
Comment (ranaaditya) [inline build.gradle:5]: use buildSrc to define new dependency versions !

File: app/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -482,4 +482,8 @@
         <item>30 mins</item>
         <item>60 mins</item>
     </string-array>
+    <string name="daily_challenge">Daily Challenges</string>
+    <string name="claim_reward">Claim Reward</string>
+    <!-- TODO: Remove or change this placeholder text -->
```
Comment (ranaaditya) [inline app/src/main/res/values/strings.xml:487]: remove this auto generated comment !

File: build.gradle [hunk 4]
```diff
@@ -11,6 +15,8 @@ buildscript {
         classpath (BuildPlugins.kotlinGradlePlugin)
         classpath (BuildPlugins.googleServicesGradlePlugin)
         classpath (BuildPlugins.firebaseCrashlyticsGradle)
+        classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version"
```
Comment (ranaaditya) [inline build.gradle:18]: use buildSrc to define gradle plugins !

**Review by ranaaditya — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/pragyan/dalal18/adapter/DailyChallengesRecyclerAdapter.kt [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): DailyChallengeViewHolder {
+        return DailyChallengeViewHolder(ChallengeListItemBinding.inflate(LayoutInflater.from(parent.context),parent,false))
+    }
+
+    override fun getItemCount(): Int = dailyChallenges.size
+
+    override fun onBindViewHolder(holder: DailyChallengeViewHolder, position: Int) {
+//
+        Log.i("daily",dailyChallenges[position].toString())
```
Comment (avikumar15) [inline app/src/main/java/org/pragyan/dalal18/adapter/DailyChallengesRecyclerAdapter.kt:32]:
```suggestion
```

File: app/src/main/java/org/pragyan/dalal18/adapter/DailyChallengesRecyclerAdapter.kt [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+
+
+        if(challengeType.equals("Cash")||challengeType.equals("NetWorth")||challengeType.equals("StockWorth"))
+            holder.binding.challengeText.text = "Increase your ${challengeType} by ${value}"
+        else{
+            val stockId = dailyChallenges[position].stockId
+            val companyName = checkUserStateListener.getCompanyNameFromStockId(stockId)
+            holder.binding.challengeText.text = "Increase the number of stocks of ${companyName} by ${value}"
+        }
+        holder.binding.rewardText.text = "\uD83D\uDCB5 $rewardAmount"
```
Comment (avikumar15) [inline app/src/main/java/org/pragyan/dalal18/adapter/DailyChallengesRecyclerAdapter.kt:46]: What is \uD83D for here?

File: app/src/main/java/org/pragyan/dalal18/adapter/DailyChallengesRecyclerAdapter.kt [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+                "Cash"->{
+                    val currentCash = checkUserStateListener.getCashWorth()
+                    progress = currentCash - initialValue
+
+                }
+                "NetWorth"->{
+                    val netWorth = checkUserStateListener.getNetWorth()
+                     progress = netWorth - initialValue
+                }
+                "StockWorth"->{
```
Comment (avikumar15) [inline app/src/main/java/org/pragyan/dalal18/adapter/DailyChallengesRecyclerAdapter.kt:64]: Make "Cash" "NetWorth" as companion objects and then use it everywhere

File: app/src/main/java/org/pragyan/dalal18/adapter/DailyChallengesRecyclerAdapter.kt [hunk 8]
```diff
@@ -0,0 +1,122 @@
...
+            }
+            holder.binding.progressTextView.text = "${progress}/${value}"
+        }else{
+            if(!userStates.isEmpty()) {
+                if (userStates[position].isCompleted && userStates[position].isRewardClamied) {
+                    holder.binding.progressTextView.visibility = View.GONE
+                    holder.binding.progressImage.visibility = View.VISIBLE
+                    holder.binding.progressImage.setImageResource(R.drawable.blue_thumb)
+                    holder.binding.claimRewardButton.visibility = View.GONE
+                } else if (userStates[position].isCompleted && !userStates[position].isRewardClamied) {
```
Comment (avikumar15) [inline app/src/main/java/org/pragyan/dalal18/adapter/DailyChallengesRecyclerAdapter.kt:87]:
```suggestion
                } else if (userStates[position].isCompleted) {
```

File: app/src/main/java/org/pragyan/dalal18/adapter/pagerAdapters/DailyChallengePagerAdapter.kt [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+    override fun getCount(): Int = totalDays
+
+    override fun getPageTitle(position: Int): CharSequence? {
+        return "Day ${position+1}"
+    }
+
+
+
+
+
```
Comment (avikumar15) [inline app/src/main/java/org/pragyan/dalal18/adapter/pagerAdapters/DailyChallengePagerAdapter.kt:23]:
```suggestion
```

File: app/src/main/java/org/pragyan/dalal18/adapter/DailyChallengesRecyclerAdapter.kt (hunk 6 above)
Comment (dhruv1294) [reply app/src/main/java/org/pragyan/dalal18/adapter/DailyChallengesRecyclerAdapter.kt:46]: Its money emoji :P

File: app/src/main/java/org/pragyan/dalal18/adapter/DailyChallengesRecyclerAdapter.kt (hunk 7 above)
Comment (dhruv1294) [reply app/src/main/java/org/pragyan/dalal18/adapter/DailyChallengesRecyclerAdapter.kt:64]: companion objects of dalalViewModel??

Comment (avikumar15) [reply app/src/main/java/org/pragyan/dalal18/adapter/DailyChallengesRecyclerAdapter.kt:64]: No, of this class itself

**Review by ranaaditya — APPROVED** [review]

**Review by avikumar15 — APPROVED** [review]

**Pull request merged**